tfjs-backend-webgpu/src/shader_util.ts

```diff
@@ -32,3 +32,23 @@ export function symbolicallyComputeStrides(
 
   return strides;
 }
+
+// atomicAdd only supports uint/int type. For float, we use
+// atomicCompareExchangeWeak to simulate.
+export const atomicAddSnippet = (dst: string, val: string) => {
+  return `
+      var oldValue = atomicLoad(&(${dst}));
+      var exchanged = false;
+      loop {
+        if exchanged {
+          break;
+        }
```

Comment:
Thanks for iterating on this, but I think the point of my previous comment was lost. The if statement should be moved to the end, otherwise we are still initializing `exchanged` here and making an unnecessary comparison before hitting `atomicCompareExchangeWeak()` for the first time.

Reply:
Done